File: 김용성/Lv2/멀쩡한 사각형.cpp
#include <stdio.h>

using namespace std;

long long gcd(long long a, long long b)
{
    while(b != 0)
    {
        long long temp = a;
        a = b;
        b = temp % b;
    }
    return a;
    
}

long long solution(int w,int h) {
    
    long long a = (long long) w;
    long long b = (long long) h;
    long long answer = 1;
    answer =  a*b - (a + b - gcd(a, b) ) ; //a*b가 int이면 overflow 발생. 따라서 Long long으로 바꿔서 하기. 
    
    return answer;
}

File: 강준서/codes/Lv.2/멀쩡한_사각형.py
import math
def solution(w,h): return w*h - w - h + math.gcd(w,h)

# 잘리는 부분이 gcd(최대공약수)만큼 반복되는 구조
# 각 패턴마다 (w/gcd + h/gcd -1)만큼 못쓰는 사각형이 생김
# 이를 정리하면 위 코드 처럼 한줄로 간단하게 나오게 된다. (유사 한줄코딩)
